### PR TITLE
Android login

### PR DESCRIPTION
**Pull request by gleono**
Implemented sign in and sign out using email and password, nothing fancy.
Also added the base structure of the app. Layouts for each of the main navigation pages was added (Tap List, Rewards and History, Settings), the first 2 are empty. We still need to add another activity for viewing an individual beer and rating it.

I was thinking on using a [ViewModel](https://developer.android.com/topic/libraries/architecture/viewmodel.html) to handle all the data received from the REST api.

Also I was thinking on using [Retrofit](http://square.github.io/retrofit/) since there are examples using both and I think it looks cleaner.

### CODE REVIEW

File: androidapp/HopSquadApp/app/google-services.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
+{
+  "project_info": {
+    "project_number": "205890785896",
+    "firebase_url": "https://hopsquadatx.firebaseio.com",
+    "project_id": "hopsquadatx",
```
Comment (rayolanderos) [inline androidapp/HopSquadApp/app/google-services.json:5]: Is there a way to connect this project to the App Engine one? I was thinking about how to implement login there that's restricted to just one user. Should we keep app / web login all in one system or separate?

File: androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/LoginActivity.java [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+    private void tryLoginOrCreateAccount(String email, final String password) {
+        if (mAuth == null) {
+            Log.d("AUTH", "Couldn't find reference to FireBaseAuth");
+            return;
+        }
+
+        final String loginEmail = email;
+        final String loginPassword = password;
+
+        mAuth.createUserWithEmailAndPassword(email, password)
```
Comment (rayolanderos) [inline androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/LoginActivity.java:150]: Does this work as both sign-in and register? It registers if you don't have an account already?

File: androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/TapListFragment.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
+package com.hopsquad.hopsquadapp;
+
+import android.app.Fragment;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+public class TapListFragment extends Fragment {
```
Comment (rayolanderos) [inline androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/TapListFragment.java:9]: What's the difference between a Fragment and an Activity?

File: androidapp/HopSquadApp/app/google-services.json (hunk 1 above)
Comment (gleono) [reply androidapp/HopSquadApp/app/google-services.json:5]: Right now the app is connected to an app engine project that I created, so I think it shouldn't be that difficult to recreate the json file with the other app engine project in firebase.

File: androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/LoginActivity.java (hunk 2 above)
Comment (gleono) [reply androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/LoginActivity.java:150]: Yes, we can change it if we want to add the create account activity too.

File: androidapp/HopSquadApp/gradlew [hunk 4]
```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env bash
```
Comment (rayolanderos) [inline androidapp/HopSquadApp/gradlew:1]: what's this fancy file?

**Review by rayolanderos — APPROVED** [review]
It looks good to me. I added a bunch of questions but nothing to update or remove for now. Mostly to clear my own ignorance :)

File: androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/TapListFragment.java (hunk 3 above)
Comment (gleono) [reply androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/TapListFragment.java:9]: Fragments are components of an android app that are attached to an activity and that are aware of its cycle. Frequently used to divide UI and logic specific to that UI in its own set of files, to later be composed in an activity. In our case the MainActivity holds a fragment at a time. The user selects an option in the bottom navigation bar and a fragment is switched into place. Each fragment is responsible of inflating its layout and responding to user input.

**Review by gwoodard3 — APPROVED** [review]

**Pull request merged**